File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlComputedColumnDefinition.cs
using System.Text;

namespace T1.SqlSharp.Expressions;

public class SqlComputedColumnDefinition : ISqlExpression
{
    public string ColumnName { get; set; } = string.Empty;
    public string Expression { get; set; } = string.Empty;
    public bool IsPersisted { get; set; } = false;
    public bool IsNotNull { get; set; } = false;
    public SqlType SqlType { get; } = SqlType.ComputedColumn;
    public TextSpan Span { get; set; } = new();

    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ComputedColumnDefinition(this);
    }

    public string ToSql()
    {
        var sql = new StringBuilder();
        sql.Append($"{ColumnName} AS {Expression}");
        if (IsPersisted)
        {
            sql.Append(" PERSISTED");
        }

        if (IsNotNull)
        {
            sql.Append(" NOT NULL");
        }

        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlComputedColumnDefinition.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlComputedColumnDefinition implements ISqlExpression {
    ColumnName: string = '';
    Expression: string = '';
    IsPersisted: boolean = false;
    IsNotNull: boolean = false;
    SqlType: SqlType = SqlType.ComputedColumn;
    Span: TextSpan = new TextSpan();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ComputedColumnDefinition(this);
    }

    ToSql(): string {
        let sql = `${this.ColumnName} AS ${this.Expression}`;
        if (this.IsPersisted) {
            sql += ' PERSISTED';
        }
        if (this.IsNotNull) {
            sql += ' NOT NULL';
        }
        return sql;
    }
} 